pop_tools/zonal_average.py

```diff
@@ -0,0 +1,212 @@
+import os
+import warnings
+
+import numpy as np
+import xarray as xr
+import xesmf as xe
+
+from .grid import get_grid
+
+
+def _generate_dest_grid(dy=None, dx=None, method_gen_grid='regular_lat_lon'):
+    """
+    Generates the destination grid
+
+    Parameters
+    ----------
+    dy: float
+      Horizontal grid spacing in y-direction (latitudinal)
+
+    dy: float
+      Horizontal grid spcaing in x-direction (longitudinal)
+    """
+
+    # Use regular lat/lon with regular spacing
+    if method_gen_grid == 'regular_lat_lon':
+        if dy is None:
+            dy = 0.25
+
+        if dx is None:
+            dx = dy
+
+    # Able to add other options at a later point
+
+    # Check to see if there is a weights file already existing
+    # Use xESMF to generate the destination grid
+    return xe.util.grid_global(dx, dy)
+
+
+def _get_default_filename(src_grid, dst_grid, method):
+
+    # Get the source grid shape
+    src_shape = src_grid.lat.shape
+
+    # Get the destination grid shape
+    dst_shape = dst_grid.lat.shape
+
+    filename = '{0}_{1}x{2}_{3}x{4}.nc'.format(
+        method, src_shape[0], src_shape[1], dst_shape[0], dst_shape[1]
+    )
```

Comment:
Feel free to ignore :) 

```suggestion
    filename = f'{method}_{src_shape[0]}x{src_shape[1]}_{dst_shape[0]}x{dst_shape[1]}.nc'
```